src/main/java/badgpt/tasks/Task.java

```diff
@@ -0,0 +1,68 @@
+package badgpt.tasks;
+
+/**
+ * Representation of a task object. The task can have a description and can be completed.
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Creates a new Task object. It is not done by default.
+     *
+     * @param description The description of the task (e.g. what must be done).
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public String getDescription() {
+        return this.description;
+    }
+
+    /**
+     * Returns the status of the task.
+     *
+     * @return [X] if it is complete. Otherwise, [ ] is returned.
+     */
+    public String getStatusIcon() {
+        return "[" + (isDone ? "X" : " ") + "]"; // mark done task with X
+    }
+
+    /**
+     * Gets the status of the task.
+     */
+    public boolean isComplete() {
+        return this.isDone;
+    }
+
+    /**
+     * Changes the status of the task to complete.
+     */
+    public void complete() {
+        this.isDone = true;
+    }
+
+    /**
+     * Changes the status of the task to incomplete.
+     */
+    public void uncomplete() {
```

Comment:
i think theres a slight spelling error here
```suggestion
    public void incomplete() {
```